packages/server/src/utilities/workerRequests.ts

```diff
@@ -167,15 +207,15 @@ export async function getChecklist(): Promise<{
 }> {
   const response = await fetch(
     checkSlashesInUrl(env.WORKER_URL + "/api/global/configs/checklist"),
-    request(undefined, { method: "GET" })
+    request({ method: "GET" })
   )
   return checkResponse(response, "get checklist")
 }
 
 export async function generateApiKey(userId: string) {
   const response = await fetch(
     checkSlashesInUrl(env.WORKER_URL + "/api/global/self/api_key"),
-    request(undefined, { method: "POST", body: { userId } })
+    request({ method: "POST", body: JSON.stringify({ userId }) })
```

Comment:
This shouldn't be necessary - the request checks if its a JSON type and sets the JSON content type header.

Reply:
If I try and pass it as an object it's a type error. I found the type in the `@types/node-fetch` library: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b2c721481ee38f65eb6630e958cc0c3383c81831/types/node-fetch/index.d.ts#L195.

An example of types not matching the reality of the library?

Reply:
ah - yes the type for node-fetch should be a string, however we use the `request` function to build a node-fetch request, and in this instance our function handles checking if the passed in body is actually JSON, sets the header and then stringifies it.

Reply:
Where does our function stringify the body?

Reply:
https://github.com/Budibase/budibase/pull/13049/files/dde8f778774ab756edb51e1c4e2a208e80b70629#r1491164920